File: trunk/server/src/main/java/org/ngame/protocol/player/CardGroupRequest.java
package org.ngame.protocol.player;

import lombok.Data;

import java.util.List;

import org.ngame.protocol.annotation.Proto;

import com.baidu.bjf.remoting.protobuf.FieldType;
import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

/**
 *
 * @author beykery
 */
@Data
@Proto(value = 8, description = "添加套牌")
public class CardGroupRequest
{
	@Protobuf(description = "套牌id")
	private String id;
	@Protobuf(description = "name")
	private String name;
	@Protobuf(description = "基地id")
	private String baseInitId;
	@Protobuf(description = "卡牌ids",fieldType=FieldType.STRING)
	private List<String> cardIds;
}

File: trunk/server/net/Protocols/CardGroupRequest.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
    /// <summary>
    /// 添加套牌 协议:8
    /// </summary>
	[Proto(value=8,description="添加套牌")]
	[ProtoContract]
	public class CardGroupRequest
	{
        /// <summary>
        ///  套牌id
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public string id;
        /// <summary>
        ///  name
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string name;
        /// <summary>
        ///  基地id
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public string baseInitId;
        /// <summary>
        ///  卡牌ids
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public List<string> cardIds;

	}
}